pkg/api/handlers/compat/images_build.go

```diff
@@ -920,6 +946,122 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func handleMultipartBuild(anchorDir string, r *http.Request) (contextDir string, additionalContexts map[string]*buildahDefine.AdditionalBuildContext, err error) {
+	reader, err := r.MultipartReader()
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to create multipart reader: %w", err)
+	}
+
+	additionalContexts = make(map[string]*buildahDefine.AdditionalBuildContext)
+	for {
+		part, err := reader.NextPart()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return "", nil, fmt.Errorf("failed to read part: %w", err)
+		}
+
+		fieldName := part.FormName()
+		fileName := part.FileName()
+
+		logrus.Debugf("Processing part: field=%q, file=%q", fieldName, fileName)
+
+		switch {
+		case fieldName == "context" && fileName != "":
+			contextPath := filepath.Join(anchorDir, "context")
+			if err := os.MkdirAll(contextPath, 0755); err != nil {
+				part.Close()
+				return "", nil, fmt.Errorf("creating context directory: %w", err)
+			}
+
+			if err := archive.Untar(part, contextPath, nil); err != nil {
```

Comment:
Should consider using `extractTarFile()` here, since it's what handles the default build context archive when the request isn't a multipart form.